frontend/src/concepts/areas/AreaContext.tsx

```diff
@@ -34,11 +36,17 @@ const AreaContextProvider: React.FC<AreaContextProps> = ({ children }) => {
   const loaded = loadedDsc && loadedDsci;
 
   const contextValue = React.useMemo(() => ({ dscStatus, dsciStatus }), [dscStatus, dsciStatus]);
-  if (error) {
+
+  if (error || !dscStatus || Object.keys(dscStatus).length === 0) {
```

Comment:
I just noticed that if the network is slow and the DSC status isn't loaded yet, we show this error screen momentarily instead of the loading screen. Can you change this condition to:
```ts
if (error || (loaded && (!dscStatus || Object.keys(dscStatus).length === 0))) {
```